packages/core/src/lib/context/fragments/renderer.svelte.ts

```diff
@@ -0,0 +1,210 @@
+  const { camera } = useCamera()
+  const { scene } = useScene()
+  const { invalidate, renderStage, autoRender, scheduler, resetFrameInvalidation } = useScheduler()
+  const { canvas, size } = useCanvas()
+
+  const renderer = options.createRenderer
+    ? options.createRenderer(canvas)
+    : new WebGLRenderer({
+        canvas,
+        powerPreference: 'high-performance',
+        antialias: true,
+        alpha: true
+      })
+
+  const autoRenderTask = renderStage.createTask(Symbol('threlte-auto-render-task'), () => {
+    renderer.render(scene, camera.current)
+  })
+
+  const context: RendererContext = {
+    renderer: renderer,
+    colorManagementEnabled: currentWritable(options.colorManagementEnabled ?? true),
+    colorSpace: currentWritable(options.colorSpace ?? 'srgb'),
+    dpr: currentWritable(options.dpr ?? window.devicePixelRatio),
+    shadows: currentWritable(options.shadows ?? PCFSoftShadowMap),
+    toneMapping: currentWritable(options.toneMapping ?? AgXToneMapping),
+    autoRenderTask
+  }
+
+  setContext<RendererContext>('threlte-renderer-context', context)
+
+  watch([context.colorManagementEnabled], ([colorManagementEnabled]) => {
+    ColorManagement.enabled = colorManagementEnabled
+  })
+
+  watch([context.colorSpace], ([colorSpace]) => {
+    if ('outputColorSpace' in renderer) {
+      renderer.outputColorSpace = colorSpace
+    }
+  })
+
+  watch([context.dpr], ([dpr]) => {
+    if ('setPixelRatio' in renderer) {
+      renderer.setPixelRatio(dpr)
+    }
+  })
+
+  // Resize the renderer when the size changes
+  const memoized = memoize(size)
+  const { start, stop } = useTask(
+    () => {
+      if (!('xr' in renderer) || renderer.xr?.isPresenting) return
+      renderer.setSize(memoized.current.width, memoized.current.height)
+      invalidate()
+      stop()
+    },
+    {
+      before: autoRenderTask,
+      autoStart: false,
+      autoInvalidate: false
+    }
+  )
+  watch([size], () => {
+    start()
+  })
+
+  watch([context.shadows], ([shadows]) => {
+    if (!('shadowMap' in renderer)) return
+
+    renderer.shadowMap.enabled = !!shadows
+    if (shadows && shadows !== true) {
+      renderer.shadowMap.type = shadows
+    } else if (shadows === true) {
+      renderer.shadowMap.type = PCFSoftShadowMap
+    }
+  })
+
+  watch([context.toneMapping], ([toneMapping]) => {
+    if (!('toneMapping' in renderer)) return
+    renderer.toneMapping = toneMapping
+  })
+
+  watch([autoRender], ([autoRender]) => {
+    if (autoRender) {
+      context.autoRenderTask.start()
+    } else {
+      context.autoRenderTask.stop()
+    }
+    return () => {
+      context.autoRenderTask.stop()
+    }
+  })
+
+  if ('setAnimationLoop' in context.renderer) {
+    const setAnimationLoop = context.renderer.setAnimationLoop as (
+      callback: (time: number) => void
+    ) => void
+    setAnimationLoop((time) => {
+      dispose()
+      scheduler.run(time)
+      resetFrameInvalidation()
+    })
+  }
```

Comment:
When is this not true?

Reply:
`context.renderer` is now extending the generic `THREE.Renderer` which currently is this:

```ts
export interface Renderer {
    domElement: HTMLCanvasElement;

    render(scene: Object3D, camera: Camera): void;
    setSize(width: number, height: number, updateStyle?: boolean): void;
}
```

Reply:
Ah!